tests/test_updown.py

```diff
@@ -4,144 +4,243 @@
+                "dd", "if=/dev/random", f"of={tmpdir}/local-file-example",
+                "bs=1M", "count=1"
+            ])
 
-    subprocess.check_call(
-        ["dd", "if=/dev/random", "of=local-file-example", "bs=1M", "count=1"])
+            with pytest.raises(HTTPError):
+                client.upload_file(
+                    f'{tmpdir}/local-file-example',
+                    'uploaded-file',
+                    chunk_size=1024**2,
+                )
 
-    with pytest.raises(ctadata.api.StorageException):
-        ctadata.upload_file('local-file-example', '../example-file')
 
+@pytest.mark.timeout(30)
+def test_webdav4_client_upload_valid(start_service):
+    with webdav_server():
+        from webdav4.client import Client
 
-def test_apiclient_upload_wrong(start_service, caplog):
-    import ctadata
-    ctadata.APIClient.downloadservice = start_service['url']
+        client = Client(start_service['url'] + "/webdav/lst")
 
-    subprocess.check_call(
-        ["dd", "if=/dev/random", "of=local-file-example", "bs=1M", "count=1"])
+        with tempfile.TemporaryDirectory() as tmpdir:
+            subprocess.check_call([
+                "dd", "if=/dev/random", f"of={tmpdir}/local-file-example",
+                "bs=1M", "count=100"
+            ])
 
-    with pytest.raises(ctadata.api.StorageException):
-        ctadata.upload_file('local-file-example',
-                            'example-files/example-file/../')
+            client.upload_file(
+                f'{tmpdir}/local-file-example',
+                'users/anonymous/uploaded-file',
+                chunk_size=1024**2,
+            )
 
 
-def test_apiclient_upload_dir(start_service, caplog):
-    import ctadata
-    ctadata.APIClient.downloadservice = start_service['url']
+@pytest.mark.timeout(30)
+def test_webdav4_client_download(start_service):
+    with webdav_server() as server:
+        subprocess.check_call([
+            "dd", "if=/dev/random",
+            f"of={server['config']['provider_mapping']['/']}/lst/file.txt",
+            "bs=1M", "count=10"
+        ])
 
-    with tempfile.TemporaryDirectory() as tmpdir:
-        for i in range(10):
-            subprocess.check_call([
-                "dd", "if=/dev/random", f"of={tmpdir}/local-file-example-{i}",
-                "bs=1M", "count=1"
-            ])
+        from webdav4.client import Client
 
-        ctadata.upload_dir(tmpdir, 'example-files/tmpdir')
+        client = Client(start_service['url'] + "/webdav/lst")
 
+        with tempfile.TemporaryDirectory() as tmpdir:
+            client.download_file(
+                'file.txt',
+                f'{tmpdir}/restored-file-example',
+            )
 
-@pytest.mark.xfail(reason="dav not implemented yet")
-def test_dav_list(start_service):
-    from webdav4.client import Client
 
-    client = Client(start_service['url'] + "/dav/lst")
-    client.exists("bla")
+@pytest.mark.timeout(30)
+def test_webdav4_client_mkdir(start_service):
```

Comment:
yeah I'd really separate webdav tests in another file too. so one extra file for ctadata and one extra for webdav?